src/wam2layers/utils/profiling.py

```diff
@@ -97,7 +96,14 @@ def print_progress(self, t, output):
             f"Time since start: {time}s, RAM: {memory:.2f} MB"
         )
 
-    def track_stability_correction(self, fy_corrected, fy_abs, config, coords, t):
+    def track_stability_correction(
+        self,
+        fy_corrected: xr.DataArray,
+        fy_abs: xr.DataArray,
+        config: Config,
+        coords: xr.Coordinates,
```

Comment:
The python3.8 tests don't like this type annotation. Should we stop supporting it?

Reply:
The type notation fails because `xr.Coordinates` was added 7 months ago, and xarray stopped supporting 3.8 a bit over a year ago. Xarray follows numpy on this, so py3.8 will use old numpy versions as well.

I think following their version restrictions would be best (i.e. deprecate support for python versions when xarray/numpy drops it)

Reply:
> I think following their version restrictions would be best (i.e. deprecate support for python versions when xarray/numpy drops it)

Sounds reasonable to me



> We probably also should specify `xarray>=2023.08.0`, that's when the coordinate object was added to the public API

I would not be very happy with that. I imagine many researchers may want to install this in an existing environment, and requiring such a recent release might lead to unnecessary conflicts. The types are not used at runtime anyway.

Reply:
> The types are not used at runtime anyway

OK but then I will have to do:
```py
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from xarray import Coordinates

def func(
        coords: "Coordinates",
):
```

otherwise it will still mess with code at runtime

Reply:
I just realized that the coords are not used anymore so there's no need for the Coordinate object.